src/ansys/turbogrid/core/multi_blade_row/multi_blade_row.py

```diff
@@ -0,0 +1,1130 @@
+        else:
+            for blade_row in self._blade_rows_to_mesh:
+                blade = self._blade_rows_to_mesh[blade_row][0]
+                if blade in rotor_blades:
+                    blade_row_spanwise_counts[blade_row] = rotor_spanwise_count
+                else:
+                    blade_row_spanwise_counts[blade_row] = stator_spanwise_count
+        return blade_row_spanwise_counts
+
+    def _get_blade_row_settings(self):
+        blade_row_settings = {}
+        for blade_row in self._blade_rows_to_mesh:
+            blade = self._blade_rows_to_mesh[blade_row][0]
+            blade_row_settings[blade_row] = []
+            if self._blade_row_gsfs is not None and blade_row in self._blade_row_gsfs:
+                blade_row_settings[blade_row].append(
+                    ("/MESH DATA", f"Global Size Factor={self._blade_row_gsfs[blade_row]}")
+                )
+            if (
+                self._blade_row_spanwise_counts is not None
+                and blade_row in self._blade_row_spanwise_counts
+            ):
+                blade_row_settings[blade_row].append(
+                    ("/MESH DATA", f"Spanwise Blade Distribution Option=Element Count and Size")
+                )
+                blade_row_settings[blade_row].append(
+                    (
+                        "/MESH DATA",
+                        f"Number Of Spanwise Blade Elements={self._blade_row_spanwise_counts[blade_row]}",
+                    )
+                )
+            if (
+                self._blade_first_element_offsets is not None
+                and blade in self._blade_first_element_offsets
+            ):
+                this_bl_f_el_off = self._blade_first_element_offsets[blade]
+                blade_row_settings[blade_row].append(
+                    (
+                        "/MESH DATA",
+                        f"Boundary Layer Specification Method=Target First Element Offset",
+                    )
+                )
+                blade_row_settings[blade_row].append(
+                    (
+                        f"/MESH DATA/EDGE SPLIT CONTROL:{blade} Boundary Layer Control",
+                        f"Target First Element Offset={this_bl_f_el_off} [m]",
+                    )
+                )
+            if self._custom_blade_settings is not None and blade in self._custom_blade_settings:
+                blade_row_settings[blade_row].extend(self._custom_blade_settings[blade])
+        return blade_row_settings
+
+    def _read_ndf(self, ndf_file, blade_row, progress_updates_queue):
+        try:
+            start_dt = dt.now()
+            ndf_name = os.path.split(ndf_file)[1]
+            ndf_name = ndf_name.split(".")[0]
+            progress_updates_queue.put([ndf_name, f"Starting {ndf_name} ndf reader"])
+            progress_updates_queue.put([ndf_name, f"Start time: {start_dt}"])
+
+            pytg_instance = launch_turbogrid(
+                product_version="24.1.0",
```

Comment:
We shouldn't hard-code the version. Why is this needed?

Reply:
I was about to remove this and then just realized that the read_tginit hotfix that was introduced recently and used by MultiBladeRow is only available since 24.1. So at least 24.1 is required. I guess there is no way to specify minimum version.

Reply:
What happens if we use read_tginit with 232? It should just be sending the action directly to TG.

Reply:
@Acohen-work, Following error occurs when product_version="23.2.0" is used:
AttributeError: 'PyTurboGrid' object has no attribute 'read_tginit'